src/FakeItEasy/ArgumentConstraintManagerExtensions.cs

```diff
@@ -301,6 +331,7 @@ public static T Matches<T>(this IArgumentConstraintManager<T> manager, Func<T, b
         [SuppressMessage("Microsoft.Design", "CA1006:DoNotNestGenericTypesInMemberSignatures", Justification = "Appropriate for Linq expressions.")]
         public static T Matches<T>(this IArgumentConstraintManager<T> scope, Expression<Func<T, bool>> predicate)
         {
+            Guard.AgainstNull(scope, nameof(scope));

```

Comment:
Maybe rename it to `manager`? The naming is quite inconsistent in those methods.